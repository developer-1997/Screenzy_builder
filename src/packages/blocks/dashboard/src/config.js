Object.defineProperty(exports, "__esModule", {
  value: true
});

// Customizable Area Start
exports.dashboardGetUrl = "/dashboard/dashboards";
exports.dashboarContentType = "application/json";
exports.dashboarApiMethodType = "GET";
exports.dashboardHost = "<calculated when request is sent>";
exports.dashboarUserAgent = "PostmanRuntime/7.26.5";
exports.dashboarAccept = "*/*";
exports.dashboarAcceptEncoding = "gzip, deflate, br";
exports.dashboarConnection = "keep-alive";
exports.dashboartoken = "";
exports.labelTitleText = "dashboard";
exports.labelBodyText = "dashboard Body";
exports.liveVideoURL =  "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerFun.mp4";
exports.offlineVideoURL = "https://samplelib.com/lib/preview/mp4/sample-10s.mp4";
exports.dummyofflineVideoURL = "https://samplelib.com/lib/preview/mp4/sample-5s.mp4";
exports.staticToken = "tyugiojpkiy8ucfghvjbuy76tfygui86t7yuhu7vbnubhhjihbjihjghjkiou897ytfyghjkiou8976tfygvhjbkiu8y76t5tfychjbuhy"
//Customizable Area End