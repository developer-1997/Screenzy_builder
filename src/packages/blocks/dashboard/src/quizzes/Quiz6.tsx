// @ts-nocheck
import React from "react";
import {
  View,
  Text,
  Image,
  TouchableOpacity,
  FlatList,
  Modal,
  BackHandler,
  ActivityIndicator,
  StyleSheet
} from "react-native";
import DashboardController from "../DashboardController";
import { CountdownCircleTimer } from "react-native-countdown-circle-timer";
import { widthPercentageToDP as wp } from "react-native-responsive-screen";

import { RFValue as rs } from "react-native-responsive-fontsize";

let timer;
let tempArray = [];

const DEFAULT_STATES = {
  count: 0,
  // Questions: [...Data],
  answerRight: "#f5f5f5",
  index: 0,
  timer: 5,
  questionAsk: true,
  win: true,
  visible: false,
  answered: false,
  key: 0
};

export default class Quiz6 extends DashboardController {
  constructor(props) {
    super(props);
    this.state = { ...DEFAULT_STATES };
  }

  componentWillMount() {
    let Quizid = this.props.navigation.state.params.quizId;

    if (Quizid) {
      this.setState({
        Quizid: Quizid,
        QuizTheme: this.props.navigation.state.params.theme
      });
      this.fetchQuizeData(Quizid);
    }
  }

  componentDidMount() {
    BackHandler.addEventListener("hardwareBackPress", this.handleBackButton);
    timer = setInterval(() => {
      /* istanbul ignore next */
      if (this.state.questionAsk) {
        if (this.state.timer >= 0) {
          this.setState({
            timer: --this.state.timer
          });
          if (this.state.timer == -1) {
            this.setState({ timer: 5 });
          }
        }
      }
    }, 1000);
  }
  componentWillUnmount() {
    BackHandler.removeEventListener("hardwareBackPress", this.handleBackButton);
    this.setState({ ...DEFAULT_STATES });
    let Quizid = this.props.navigation.state.params.quizId;
    this.saveUserQuizAnswers(Quizid, tempArray);

    setTimeout(() => {
      tempArray.length = 0;
    }, 500);
  }

  MoveToNextQuestion = nextIndex => {
    this.setState({ index: nextIndex });
  };
  quizLostSix = () => {
    this.setState({ visible: true });
    setTimeout(() => {
      this.setState({ visible: false, win: true });
      this.props.navigation.navigate("Home");
    }, 3000);
    this.setState({ videoPlay: false });
  };

  handleBackButton() {
    return true;
  }

  handelOnComplete = () => {
    if (
      this.state.index + 1 >=
      this.state.quizQuestionData.attributes.questions.length
    ) {
      let temp = { ...this.state.quizQuestionData };
      if (temp.attributes.questions[this.state.index].Answered == undefined) {
        let tempobj = {
          question_id: temp.attributes.questions[this.state.index].id,
          option_id: null
        };
        tempArray.push(tempobj);
      }

      if (
        temp.attributes.questions[this.state.index].Answered &&
        this.state.win
      ) {
        clearInterval(timer);
        this.props.navigation.replace("Win", {
          id: this.props.navigation.state.params.quizId
        });
      } else {
        clearInterval(timer);
        this.quizLostSix();
      }
    } else {
      let temp = { ...this.state.quizQuestionData };
      if (temp.attributes.questions[this.state.index].Answered == undefined) {
        this.setState({ win: false });
        let tempobj = {
          question_id: temp.attributes.questions[this.state.index].id,
          option_id: null
        };
        tempArray.push(tempobj);
      }

      this.setState({
        quizQuestionData: temp
      });

      this.MoveToNextQuestion(this.state.index + 1);
      return { shouldRepeat: true, delay: 1 };
    }
  };

  handelOptionPress = (item, index) => {
    let temp = { ...this.state.quizQuestionData };

    for (
      let i = 0;
      i < temp.attributes.questions[this.state.index].attributes.options.length;
      i++
    ) {
      let CurrentOption =
        temp.attributes.questions[this.state.index].attributes.options[i];
      if (CurrentOption.id == item.id) {
        if (CurrentOption.attributes.is_correct) {
          CurrentOption.bg = "green";
          this.setState({
            count: this.state.count + 1,
            answered: true
          });
          if (
            this.state.index + 1 >=
            this.state.quizQuestionData.attributes.questions.length
          ) {
            if (this.state.win) {
              this.props.navigation.replace("Win", {
                id: this.props.navigation.state.params.quizId
              });
            } else {
              this.quizLostSix();
            }
          } else {
            this.MoveToNextQuestion(this.state.index + 1);
            this.setState({ key: this.state.key + 1 });
          }
        } else {
          this.setState({ win: false, answered: true });
          CurrentOption.bg = "red";
          if (
            this.state.index + 1 >=
            this.state.quizQuestionData.attributes.questions.length
          ) {
            this.quizLostSix();
          } else {
            this.MoveToNextQuestion(this.state.index + 1);
            this.setState({ key: this.state.key + 1 });
          }
        }
      } else {
        CurrentOption.bg = null;
      }
      temp.attributes.questions[this.state.index].attributes.options[
        i
      ] = CurrentOption;
    }
    temp.attributes.questions[this.state.index].Answered = true;

    this.setState({
      quizQuestionData: temp
    });
    let tempobj = {
      question_id: temp.attributes.questions[this.state.index].id,
      option_id:
        temp.attributes.questions[this.state.index].attributes.options[index].id
    };
    tempArray.push(tempobj);
  };

  render() {
    const colorMap = {
      white: "#000",
      black: "#fff"
    };

    const csColorQuiz6 = colorMap[this.state.QuizTheme] || "grey";
    if (!this.state.quizQuestionLoaded) {
      return (
        <View style={styles.wrapperQuizSix}>
          <Image
            source={require("../../assets/mainlogo.png")}
            style={styles.mainLogoQuizSix}
          />
          <ActivityIndicator size={"large"} color="#fff" />
        </View>
      );
    }
    return (
      <View testID="showQuiz" style={styles.showQuizSix}>
        <Image
          source={require("../../assets/mainlogo.png")}
          style={styles.mainLogoQuizSix}
        />
        <Modal
          style={{}}
          transparent={true}
          visible={this.state.visible}
          onRequestClose={() => {}}
        >
          <View style={styles.modalWrapperOneQuizSix}>
            <View style={styles.modalWrapperTwoQuizSix}>
              <Image
                source={require("../../assets/networkloss.png")}
                style={styles.lostIconQuizSix}
              />
              <Text style={styles.quizSixLostTextOne}>Oh No!!</Text>
              <View style={{ padding: rs(10) }}>
                <Text style={styles.quizSixLostTextTwo}>
                  Looks like you lost, Better Luck Next Time.
                </Text>
              </View>
            </View>
          </View>
        </Modal>

        <View
          style={{
            width: rs(380),

            backgroundColor: this.state.QuizTheme
              ? this.state.QuizTheme
              : "#f5f5f5",
            borderRadius: rs(5),
            alignItems: "center",
            padding: rs(10)
          }}
        >
          <View
            style={{
              flexDirection: "row",
              alignItems: "center",
              justifyContent: "space-between"
            }}
          >
            <View
              style={{
                width: rs(20),
                height: rs(20),
                borderRadius: rs(10),
                backgroundColor: this.state.timer < 5 ? "grey" : "#8F00FF",

                borderWidth: rs(0.5),
                borderColor: "orange"
              }}
            />
            <View
              style={{
                width: rs(20),
                height: rs(20),
                borderRadius: rs(10),
                backgroundColor: this.state.timer < 4 ? "grey" : "#8F00FF",
                marginLeft: rs(10),
                borderWidth: rs(0.5),
                borderColor: "orange"
              }}
            />
            <View
              style={{
                width: rs(20),
                height: rs(20),
                borderRadius: rs(10),
                backgroundColor: this.state.timer < 3 ? "grey" : "#8F00FF",
                marginLeft: rs(10),
                borderWidth: rs(0.5),
                borderColor: "orange"
              }}
            />
            <View
              style={{
                width: rs(20),
                height: rs(20),
                borderRadius: rs(10),
                backgroundColor: this.state.timer < 2 ? "grey" : "#8F00FF",
                marginLeft: rs(10),
                borderWidth: rs(0.5),
                borderColor: "orange"
              }}
            />
            <View
              style={{
                width: rs(20),
                height: rs(20),
                borderRadius: rs(10),
                backgroundColor: this.state.timer < 1 ? "grey" : "#8F00FF",
                marginLeft: rs(10),
                borderWidth: rs(0.5),
                borderColor: "orange"
              }}
            />
            <View
              style={{
                width: rs(40),
                height: rs(40),
                borderRadius: rs(20),
                backgroundColor: "orange",
                marginHorizontal: rs(8),
                marginBottom: rs(15)
              }}
            >
              <CountdownCircleTimer
                key={this.state.key}
                strokeWidth={5}
                trailColor={"#fff"}
                size={rs(40)}
                rotation={"counterclockwise"}
                isPlaying
                duration={
                  this.props.navigation.state.params.question_Time
                    ? this.props.navigation.state.params.question_Time
                    : 5
                }
                colors={["#8F00FF", "#8F00FF", "#8F00FF", "#8F00FF"]}
                colorsTime={[5, 2, 0]}
                onComplete={this.handelOnComplete}
              >
                {({ remainingTime }) => (
                  <Text style={{ color: "#fff", fontSize: rs(14) }}>
                    {remainingTime < 10 ? "0" : null}
                    {remainingTime}
                  </Text>
                )}
              </CountdownCircleTimer>
            </View>
            <View
              style={{
                width: rs(20),
                height: rs(20),
                borderRadius: rs(10),
                backgroundColor: this.state.timer < 1 ? "grey" : "#8F00FF",

                borderWidth: rs(0.5),
                borderColor: "orange"
              }}
            />
            <View
              style={{
                width: rs(20),
                height: rs(20),
                borderRadius: rs(10),
                backgroundColor: this.state.timer < 2 ? "grey" : "#8F00FF",
                marginLeft: rs(10),
                borderWidth: rs(0.5),
                borderColor: "orange"
              }}
            />
            <View
              style={{
                width: rs(20),
                height: rs(20),
                borderRadius: rs(10),
                backgroundColor: this.state.timer < 3 ? "grey" : "#8F00FF",
                marginLeft: rs(10),
                borderWidth: rs(0.5),
                borderColor: "orange"
              }}
            />
            <View
              style={{
                width: rs(20),
                height: rs(20),
                borderRadius: rs(10),
                backgroundColor: this.state.timer < 4 ? "grey" : "#8F00FF",
                marginLeft: rs(10),
                borderWidth: rs(0.5),
                borderColor: "orange"
              }}
            />
            <View
              style={{
                width: rs(20),
                height: rs(20),
                borderRadius: rs(10),
                backgroundColor: this.state.timer < 5 ? "grey" : "#8F00FF",
                marginLeft: rs(10),
                borderWidth: rs(0.5),
                borderColor: "orange"
              }}
            />
          </View>
          <Text
            style={{
              color: csColorQuiz6,
              fontSize: rs(14),
              marginTop: rs(10),
              fontFamily: "Montserrat-SemiBold"
            }}
          >
            {
              this.state.quizQuestionData.attributes.questions[this.state.index]
                .attributes.title
            }
          </Text>
          <FlatList
            numColumns={2}
            data={
              this.state.quizQuestionData.attributes.questions[this.state.index]
                .attributes.options
            }
            extraData={this.state.Questions}
            keyExtractor={item => item.id}
            renderItem={({ item, index }) => {
              console.log(item.answer, "flatlist");
              return (
                <TouchableOpacity
                  onPress={() => this.handelOptionPress(item, index)}
                  style={{
                    backgroundColor: item && item.bg ? item.bg : "#f5f5f5",
                    padding: rs(10),
                    elevation: rs(5),
                    marginTop: rs(10),
                    borderRadius: rs(5),
                    width: rs(120),
                    alignItems: "center",
                    marginLeft: rs(10)
                  }}
                  disabled={
                    this.state.quizQuestionData.attributes.questions[
                      this.state.index
                    ].Answered
                  }
                >
                  <Image
                    source={{ uri: item.attributes.option_image }}
                    style={{ width: rs(50), height: rs(50) }}
                  />
                  <Text
                    style={{
                      color: "#000",
                      fontSize: rs(12),
                      fontFamily: "Montserrat-Medium",
                      textAlign: "center"
                    }}
                  >
                    {item.attributes.option_title}
                  </Text>
                </TouchableOpacity>
              );
            }}
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  wrapperQuizSix: {
    flex: 1,
    backgroundColor: "#000",
    justifyContent: "center",
    alignItems: "center"
  },
  mainLogoQuizSix: {
    width: rs(100),
    height: rs(100),
    resizeMode: "contain"
  },
  showQuizSix: {
    flex: 1,
    backgroundColor: "#1c1616",
    justifyContent: "center",
    alignItems: "center"
  },
  showQuizIconSix: {
    width: rs(100),
    height: rs(100),
    resizeMode: "contain",
    position: "absolute",
    left: 15,
    top: -25
  },
  modalWrapperOneQuizSix: {
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "rgba(0,0,0,0.5)",
    width: "100%",
    height: "100%"
  },
  modalWrapperTwoQuizSix: {
    backgroundColor: "#fff",
    width: wp(40),
    height: rs(150),
    borderRadius: rs(10)
  },
  lostIconQuizSix: {
    width: rs(70),
    height: rs(70),
    position: "absolute",
    alignSelf: "center",
    top: rs(-50)
  },
  quizSixLostTextOne: {
    fontSize: rs(22),
    fontFamily: "Montserrat-SemiBold",
    textAlign: "center",
    marginTop: rs(30)
  },
  quizSixLostTextTwo: {
    fontSize: rs(16),
    fontFamily: "Montserrat-Medium",
    textAlign: "center"
  }
});
