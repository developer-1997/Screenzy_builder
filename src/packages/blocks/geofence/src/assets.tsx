export const AddIcon = require("../assets/add.svg") 