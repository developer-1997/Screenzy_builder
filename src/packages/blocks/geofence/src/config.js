Object.defineProperty(exports, "__esModule", {
  value: true
});

// Customizable Area Start
exports.labelTitleText = "Geofence";
exports.labelBodyText = "Geofence Body";
exports.radarKey = "prj_live_pk_dc7bbaa26e61343e20a955965be95a7035dd611a";
exports.geofenceListApi="/bx_block_geofence2/geofence" ;
exports.geofenceListMethod="GET"
exports.geofenceContentType="application/json";
exports.geofenceListTitle="List of geofences"
exports.geofenceAddTitle="Add"
exports.editTitle="Editor"
exports.backButtonTitle="Back"
exports.geofenceDeleteMethodType="DELETE";
exports.geofenceUpdateMethodType="PUT"
exports.geofenceCreateMethodType="POST"
// Customizable Area End