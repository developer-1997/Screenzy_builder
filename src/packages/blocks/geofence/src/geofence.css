html,
body {
  padding: 0;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen,
    Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
}

.container {
  display: flex;
  width: 100vw;
  height: 100vh;
  overflow: hidden;
}

input.error {
  border: 3px solid red;
}

.controls {
  overflow-y: scroll;
  overflow-x: hidden;
  padding: 0 6px;
  position: relative;
  flex: 3;
  max-width: 20rem;
  background: #14161a;
  color: #fff;
  border-right: 0.5px solid gray;
}
.controls .form {
  display: flex;
  flex-direction: column;
}
.controls .form > * {
  margin-bottom: 12px;
}
.controls h1,
h2,
h3 {
  padding: 1rem;
}
.showcase {
  width: 100%;
  min-height: 100vh;
}

.fence-box {
  padding: 2px 12px;
  border-radius: 6px;
  margin-bottom: 12px;
  width: 100%;
  font-size: large;
  display: flex;
  justify-content: space-between;
}

.fence-box.active {
  background-color: darkorange;
}

.fence-box.inactive {
  background-color: gray;
}

.fence-box .edit {
  display: none;
  font-size: small;
  padding: 4px 1rem;
}

.fence-box:hover .edit {
  display: block;
}

.controls input {
  border: none;
  padding: 1rem;
  border-radius: 8px;
}

.manual-control {
  margin-top: 2rem;
}

.manual-control > * {
  margin-bottom: 1.5rem;
}

.map-container {
  width: 100%;
  height: 100vh;
}

.back-button {
  margin: 8px 4px;
}

.map {
  flex: 7;
  height: 100vh;
}

@media screen and (max-width: 600px) {
  .container {
    flex-direction: column;
  }
  .map,
  .map-container {
    height: 50vh;
    flex: 5;
  }
  .controls {
    max-width: 100%;
    height: 50vh;
    flex: 5;
  }
  .controls .map-not-available {
    display: block;
  }
}

.highlight {
  font-size: 1.25rem;
  font-weight: bold;
}

.combobox-input {
  width: 100%;
  padding: 0.5rem;
}


:root {
  font-family: Inter, Avenir, Helvetica, Arial, sans-serif;
  font-size: 16px;
  line-height: 24px;
  font-weight: 400;

  color-scheme: light dark;
  color: rgba(255, 255, 255, 0.87);
  background-color: #242424;

  font-synthesis: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-size-adjust: 100%;
}

a {
  font-weight: 500;
  color: #646cff;
  text-decoration: inherit;
}
a:hover {
  color: #535bf2;
}

body {
  margin: 0;
  display: flex;
  place-items: center;
  min-width: 320px;
  min-height: 100vh;
}

h1 {
  font-size: 3.2em;
  line-height: 1.1;
}

button {
  border-radius: 8px;
  border: 1px solid transparent;
  padding: 0.6em 1.2em;
  font-size: 1em;
  font-weight: 500;
  font-family: inherit;
  background-color: #1a1a1a;
  cursor: pointer;
  transition: border-color 0.25s;
}
button:hover {
  border-color: #646cff;
}
button:focus,
button:focus-visible {
  outline: 4px auto -webkit-focus-ring-color;
}

@media (prefers-color-scheme: light) {
  :root {
    color: #213547;
    background-color: #ffffff;
  }
  a:hover {
    color: #747bff;
  }
  button {
    background-color: #f9f9f9;
  }
}
