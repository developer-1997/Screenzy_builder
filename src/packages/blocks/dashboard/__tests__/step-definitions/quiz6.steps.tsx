import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";
import * as helpers from "../../../../framework/src/Helpers";
import React from "react";
import Quiz6 from "../../src/quizzes/Quiz6";
import { runEngine } from "../../../../framework/src/RunEngine";
import { Message } from "../../../../framework/src/Message";
import MessageEnum, {
  getName
} from "../../../../framework/src/Messages/MessageEnum";
import { View } from "react-native";

const navigation = require("react-navigation");
navigation.replace = jest.fn();
const mockNavigation = {
  state: {
    params: {
      Quizid: "136"
    }
  },
  replace: jest.fn()
};
const screenProps = {
  navigation: mockNavigation,
  id: "Quiz6"
};
const feature = loadFeature("./__tests__/features/quiz6-scenario.feature");

const quiz6QuestionDataMock = {
  id: "136",
  type: "quiz",
  attributes: {
    theme: "black",
    quiz_type: "Quiz-6",
    questions: [
      {
        id: "184",
        type: "question",
        attributes: {
          question_type: "title",
          title:
            "Which water animal has the strongest smelling ability? (Answer and win 30 Rs Amazon voucher)",
          question_image: null,
          options: [
            {
              id: "644",
              type: "option",
              attributes: {
                option_type: "title",
                option_title: "Shark",
                is_correct: false,
                option_image: null
              }
            },
            {
              id: "645",
              type: "option",
              attributes: {
                option_type: "title",
                option_title: "Whale",
                is_correct: true,
                option_image: null
              }
            },
            {
              id: "646",
              type: "option",
              attributes: {
                option_type: "title",
                option_title: "Octopus",
                is_correct: false,
                option_image: null
              }
            }
          ]
        }
      }
    ]
  }
};

defineFeature(feature, test => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "web" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
  });

  test("User navigates to quiz6", ({ given, when, then }) => {
    let quiz6Wrapper: ShallowWrapper;
    let instance: Quiz6;
    given("I am a User loading quiz6", () => {
      quiz6Wrapper = shallow(<Quiz6 {...screenProps} />);
      expect(quiz6Wrapper).toBeTruthy();
      expect(quiz6Wrapper).toMatchSnapshot();
    });

    when("I navigate to the quiz6", () => {
      instance = quiz6Wrapper.instance() as Quiz6;
      expect(quiz6Wrapper).toBeTruthy();
      expect(quiz6Wrapper).toMatchSnapshot();
    });
    then("quiz6 will load with out errors", () => {
      expect(quiz6Wrapper).toBeTruthy();
      expect(quiz6Wrapper).toMatchSnapshot();
    });

    then("Quiz6 will display messages", () => {
      const tokenMsg: Message = new Message(
        getName(MessageEnum.SessionResponseMessage)
      );
      tokenMsg.addData(getName(MessageEnum.SessionResponseToken), "TOKEN");
      runEngine.sendMessage("Unit Test", tokenMsg);

      const apiMsg: Message = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      apiMsg.addData(getName(MessageEnum.RestAPIResponceSuccessMessage), {
        data: [
          {
            id: 1,
            type: "quiz6",
            attributes: {
              id: 1,
              title: "Quiz6_title_1",
              value: 1,
              created_at: "2023-04-02T17:10:08.139Z",
              updated_at: "2023-04-02T17:10:08.139Z"
            }
          },
          {
            id: 2,
            type: "quiz6",
            attributes: {
              id: 2,
              title: "Quiz6 5",
              value: 5,
              created_at: "2023-04-02T17:10:36.867Z",
              updated_at: "2023-04-02T17:10:36.867Z"
            }
          }
        ]
      });
      runEngine.sendMessage("Unit Test", apiMsg);

      expect(quiz6Wrapper).toBeTruthy();
      expect(quiz6Wrapper).toMatchSnapshot();
    });
    then("Quiz6 will display notifcation if no messages", () => {
      const apiNoItemsMsg: Message = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      apiNoItemsMsg.addData(
        getName(MessageEnum.RestAPIResponceSuccessMessage),
        { data: [] }
      );
      runEngine.sendMessage("Unit Test", apiNoItemsMsg);

      expect(quiz6Wrapper).toBeTruthy();
      expect(quiz6Wrapper).toMatchSnapshot();
    });
    then("Quiz6 will display notifcation if API failure", () => {
      const apiNoItemsMsg: Message = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      apiNoItemsMsg.addData(
        getName(MessageEnum.RestAPIResponceSuccessMessage),
        { data: [] }
      );
      runEngine.sendMessage("Unit Test", apiNoItemsMsg);

      const apiErrorResponceMsg: Message = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      apiErrorResponceMsg.addData(
        getName(MessageEnum.RestAPIResponceSuccessMessage),
        { errors: "Error" }
      );
      runEngine.sendMessage("Unit Test", apiErrorResponceMsg);

      const apiFailedErrorResponceMsg: Message = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      runEngine.sendMessage("Unit Test", apiFailedErrorResponceMsg);

      expect(quiz6Wrapper).toBeTruthy();
      expect(quiz6Wrapper).toMatchSnapshot();
    });

    then("I can leave the screen with out errors", () => {
      instance.componentWillUnmount();
      expect(quiz6Wrapper).toBeTruthy();
      expect(quiz6Wrapper).toMatchSnapshot();
    });
  });

  test("test for show quiz modal", ({ given, when, then }) => {
    let quizWrapper: ShallowWrapper;
    let instance: Quiz6;

    given("I am a User attempting to quiz", () => {
      quizWrapper = shallow(<Quiz6 {...screenProps} />);
      expect(quizWrapper).toBeTruthy();
    });

    when("I am user try to attempting quiz", () => {
      instance = quizWrapper.instance() as Quiz6;
      instance.setState({ quizQuestionLoaded: true });
    });

    then("I can attempt the quiz", () => {
      expect(quizWrapper.contains(<View testID="showQuiz" />)).toBe(false);
    });
  });

  test("test for you win quiz6", ({ given, when, then }) => {
    let quizWrapper1: ShallowWrapper;
    let instance: Quiz6;

    given("I am a User navigate to quiz6 win", () => {
      quizWrapper1 = shallow(<Quiz6 {...screenProps} />);
      expect(quizWrapper1).toBeTruthy();
    });

    when("I am user try to load quiz6 win", () => {
      instance = quizWrapper1.instance() as Quiz6;
      instance.setState({
        quizQuestionData: quiz6QuestionDataMock,
        index: 0
      });
    });

    then("I win the quiz6", () => {
      instance.handelOnComplete();
    });
  });

  test("test for you loss quiz6", ({ given, when, then }) => {
    let quizWrapper1: ShallowWrapper;
    let instance: Quiz6;

    given("I am a User navigate to quiz6 loss", () => {
      quizWrapper1 = shallow(<Quiz6 {...screenProps} />);
      expect(quizWrapper1).toBeTruthy();
    });

    when("I am user try to load quiz6 loss", () => {
      instance = quizWrapper1.instance() as Quiz6;
      instance.setState({
        index: 0
      });
    });

    then("I loss the quiz6", () => {
      instance.handelOnComplete();
    });
  });

  test("test for clicked quiz6 options", ({ given, when, then }) => {
    let quizWrapper1: ShallowWrapper;
    let instance: Quiz6;

    given("I can see quiz6 options", () => {
      quizWrapper1 = shallow(<Quiz6 {...screenProps} />);
      expect(quizWrapper1).toBeTruthy();
    });

    when("I am user try to click one oaption of quiz6", () => {
      instance = quizWrapper1.instance() as Quiz6;
      instance.setState({
        index: 0,
        timerClock: 15,
        quizQuestionData: quiz6QuestionDataMock
      });
      instance.handelOnComplete();
    });

    then("I  clicked wrong  answer", () => {
      instance.handelOptionPress(
        quiz6QuestionDataMock.attributes.questions[0].attributes.options[0],
        0
      );
      expect(quizWrapper1).toBeTruthy();
    });

    then("I can clicked right answer", () => {
      instance.handelOptionPress(
        quiz6QuestionDataMock.attributes.questions[0].attributes.options[1],
        -2
      );
    });

    then("I can clicked right answer 2", () => {
      instance.handelOptionPress(
        quiz6QuestionDataMock.attributes.questions[0].attributes.options[2],
        -2
      );
    });
  });
});
