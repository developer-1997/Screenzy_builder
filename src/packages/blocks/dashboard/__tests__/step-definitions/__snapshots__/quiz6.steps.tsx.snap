// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quiz6 User navigates to quiz6 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../packages/blocks/dashboard/assets/mainlogo.png",
      }
    }
    style={
      Object {
        "height": 196,
        "resizeMode": "contain",
        "width": 196,
      }
    }
  />
  <ActivityIndicator
    color="#fff"
    size="large"
  />
</View>
`;

exports[`quiz6 User navigates to quiz6 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../packages/blocks/dashboard/assets/mainlogo.png",
      }
    }
    style={
      Object {
        "height": 196,
        "resizeMode": "contain",
        "width": 196,
      }
    }
  />
  <ActivityIndicator
    color="#fff"
    size="large"
  />
</View>
`;

exports[`quiz6 User navigates to quiz6 3`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../packages/blocks/dashboard/assets/mainlogo.png",
      }
    }
    style={
      Object {
        "height": 196,
        "resizeMode": "contain",
        "width": 196,
      }
    }
  />
  <ActivityIndicator
    color="#fff"
    size="large"
  />
</View>
`;

exports[`quiz6 User navigates to quiz6 4`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../packages/blocks/dashboard/assets/mainlogo.png",
      }
    }
    style={
      Object {
        "height": 196,
        "resizeMode": "contain",
        "width": 196,
      }
    }
  />
  <ActivityIndicator
    color="#fff"
    size="large"
  />
</View>
`;

exports[`quiz6 User navigates to quiz6 5`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../packages/blocks/dashboard/assets/mainlogo.png",
      }
    }
    style={
      Object {
        "height": 196,
        "resizeMode": "contain",
        "width": 196,
      }
    }
  />
  <ActivityIndicator
    color="#fff"
    size="large"
  />
</View>
`;

exports[`quiz6 User navigates to quiz6 6`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../packages/blocks/dashboard/assets/mainlogo.png",
      }
    }
    style={
      Object {
        "height": 196,
        "resizeMode": "contain",
        "width": 196,
      }
    }
  />
  <ActivityIndicator
    color="#fff"
    size="large"
  />
</View>
`;

exports[`quiz6 User navigates to quiz6 7`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../packages/blocks/dashboard/assets/mainlogo.png",
      }
    }
    style={
      Object {
        "height": 196,
        "resizeMode": "contain",
        "width": 196,
      }
    }
  />
  <ActivityIndicator
    color="#fff"
    size="large"
  />
</View>
`;
